// spinners-angular/fesm2015/spinners-angular-spinner-circular-fixed.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

class SpinnerCircularFixedComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 4 * (this.thickness / 100);
    }
    get dashStyle() {
        return Object.assign({ color: this.color }, (!this.still
            ? { animation: `spinners-angular-circular-fixed ${140 / this.speed}s linear infinite` }
            : {}));
    }
}
SpinnerCircularFixedComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-circular-fixed',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"40, 134\"\r\n    stroke-dashoffset=\"325\"\r\n    stroke-linecap=\"round\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-circular-fixed{0%{stroke-dashoffset:325}to{stroke-dashoffset:151}}@keyframes spinners-angular-circular-fixed{0%{stroke-dashoffset:325}to{stroke-dashoffset:151}}\n"]
            },] }
];

class SpinnerCircularFixedModule {
}
SpinnerCircularFixedModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [SpinnerCircularFixedComponent],
                exports: [SpinnerCircularFixedComponent],
            },] }
];

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerCircularFixedComponent, SpinnerCircularFixedModule };
