// spinners-angular/fesm2015/spinners-angular-spinner-circular-split.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

class SpinnerCircularSplitComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 4 * (this.thickness / 100);
    }
    get dashStyle() {
        return Object.assign({ color: this.color }, (!this.still
            ? { animation: `spinners-angular-circular-split ${140 / this.speed}s linear infinite` }
            : {}));
    }
}
SpinnerCircularSplitComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-circular-split',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"5, 170\"\r\n    stroke-dashoffset=\"1\"\r\n    stroke-linecap=\"round\"\r\n    transform=\"rotate(-90 33 33)\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-circular-split{0%{stroke-dashoffset:1;stroke-dasharray:5,170}10%{stroke-dashoffset:13;stroke-dasharray:30,145}13%{stroke-dashoffset:-11;stroke-dasharray:5,145}50%{stroke-dasharray:5,0,5,165;stroke-dashoffset:-82}51%{stroke-dasharray:2,0,2,139;stroke-dashoffset:-85}61%{stroke-dasharray:15,0,15,165;stroke-dashoffset:-72}64%{stroke-dasharray:5,20,5,145;stroke-dashoffset:-72}}@keyframes spinners-angular-circular-split{0%{stroke-dashoffset:1;stroke-dasharray:5,170}10%{stroke-dashoffset:13;stroke-dasharray:30,145}13%{stroke-dashoffset:-11;stroke-dasharray:5,145}50%{stroke-dasharray:5,0,5,165;stroke-dashoffset:-82}51%{stroke-dasharray:2,0,2,139;stroke-dashoffset:-85}61%{stroke-dasharray:15,0,15,165;stroke-dashoffset:-72}64%{stroke-dasharray:5,20,5,145;stroke-dashoffset:-72}}\n"]
            },] }
];

class SpinnerCircularSplitModule {
}
SpinnerCircularSplitModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [SpinnerCircularSplitComponent],
                exports: [SpinnerCircularSplitComponent],
            },] }
];

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerCircularSplitComponent, SpinnerCircularSplitModule };
