// spinners-angular/fesm2015/spinners-angular-spinners-utils.js
import { Input } from '@angular/core';

class SpinnerComponent {
    constructor() {
        this.color = '#38ad48';
        this.enabled = true;
        this.size = 50;
        this.speed = 100;
        this.still = false;
        this.styles = {};
        this.thickness = 100;
    }
    get svgStyle() {
        return Object.assign({ color: this.color, overflow: 'visible', width: normalizeSize(this.size), height: normalizeSize(this.size) }, (typeof this.styles === 'string' ? JSON.parse(this.styles) : this.styles));
    }
}
SpinnerComponent.propDecorators = {
    color: [{ type: Input }],
    enabled: [{ type: Input }],
    size: [{ type: Input }],
    speed: [{ type: Input }],
    still: [{ type: Input }],
    styles: [{ type: Input }],
    thickness: [{ type: Input }]
};
class SpinnerComponentWithSecondaryColor extends SpinnerComponent {
    constructor() {
        super(...arguments);
        this.secondaryColor = 'rgba(0,0,0,0.44)';
    }
}
SpinnerComponentWithSecondaryColor.propDecorators = {
    secondaryColor: [{ type: Input }]
};
const normalizeSize = (size) => (parseFloat(size.toString()).toString() === size.toString()
    ? `${size}px`
    : size.toString());

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerComponent, SpinnerComponentWithSecondaryColor, normalizeSize };
