// spinners-angular/fesm2015/spinners-angular-spinner-diamond.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

class SpinnerDiamondComponent extends SpinnerComponentWithSecondaryColor {
    constructor() {
        super(...arguments);
        this.colorCoords = [
            { x: 3, y: 48 },
            { x: 18, y: 33 },
            { x: 18, y: 48 },
            { x: 18, y: 63 },
            { x: 33, y: 48 },
        ];
    }
    get coords() {
        return [
            ...this.colorCoords,
            { x: 33, y: 18 },
            { x: 33, y: 33 },
            { x: 33, y: 63 },
            { x: 33, y: 78 },
            { x: 49, y: 3 },
            { x: 49, y: 18 },
            { x: 49, y: 33 },
            { x: 49, y: 48 },
            { x: 49, y: 63 },
            { x: 49, y: 78 },
            { x: 49, y: 93 },
            { x: 64, y: 18 },
            { x: 64, y: 33 },
            { x: 64, y: 48 },
            { x: 64, y: 63 },
            { x: 64, y: 78 },
            { x: 79, y: 33 },
            { x: 79, y: 48 },
            { x: 79, y: 63 },
            { x: 94, y: 48 },
        ];
    }
    get groupStyle() {
        return {
            animation: !this.still ? `spinners-angular-diamond ${140 / this.speed}s steps(2, end) infinite` : 'none',
        };
    }
}
SpinnerDiamondComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-diamond',
                template: "<svg *ngIf=\"enabled\" fill=\"currentColor\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 96 96\">\r\n  <circle\r\n    *ngFor=\"let c of coords\"\r\n    [attr.cx]=\"c.x\"\r\n    [attr.cy]=\"c.y\"\r\n    [attr.r]=\"2.5 * (thickness / 100)\"\r\n    [attr.fill]=\"secondaryColor\"\r\n  />\r\n  <g [ngStyle]=\"groupStyle\">\r\n    <circle\r\n      *ngFor=\"let c of colorCoords; index as i\"\r\n      [attr.r]=\"3.5 * (thickness / 100)\"\r\n      [attr.cx]=\"c.x\"\r\n      [attr.cy]=\"c.y\"\r\n    />\r\n  </g>\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-diamond{25%{transform:translate(30px,-30px)}50%{transform:translate(60px)}75%{transform:translate(30px,30px)}}@keyframes spinners-angular-diamond{25%{transform:translate(30px,-30px)}50%{transform:translate(60px)}75%{transform:translate(30px,30px)}}\n"]
            },] }
];

class SpinnerDiamondModule {
}
SpinnerDiamondModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [SpinnerDiamondComponent],
                exports: [SpinnerDiamondComponent],
            },] }
];

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerDiamondComponent, SpinnerDiamondModule };
