// spinners-angular/fesm2015/spinners-angular-spinner-infinity.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

class SpinnerInfinityComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 7 * (this.thickness / 100);
    }
    get dashStyle() {
        return !this.still
            ? { animation: `spinners-angular-infinity ${140 / this.speed}s linear infinite` }
            : {};
    }
}
SpinnerInfinityComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-infinity',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 131 55\">\r\n  <defs>\r\n    <path\r\n      d=\"M46.57 45.5138C36.346 55.4954 19.8919 55.4954 9.66794 45.5138C-0.55598 35.5321 -0.55598 19.4678 9.66794 9.48624C19.8919 -0.495412 36.346 -0.495412 46.57 9.48624L84.4303 45.5138C94.6543 55.4954 111.108 55.4954 121.332 45.5138C131.556 35.5321 131.556 19.4678 121.332 9.48624C111.108 -0.495412 94.6543 -0.495412 84.4303 9.48624L46.57 45.5138Z\"\r\n      id=\"spinners-react-infinity-path\"\r\n    />\r\n  </defs>\r\n  <use\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    xlink:href=\"#spinners-react-infinity-path\"\r\n  />\r\n  <use\r\n    fill=\"none\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"1, 347\"\r\n    stroke-dashoffset=\"75\"\r\n    stroke-linecap=\"round\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n    xlink:href=\"#spinners-react-infinity-path\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-infinity{0%{stroke-dasharray:1,347;stroke-dashoffset:75}25%,75%{stroke-dasharray:17,330}50%{stroke-dasharray:1,347}to{stroke-dasharray:1,347;stroke-dashoffset:423}}@keyframes spinners-angular-infinity{0%{stroke-dasharray:1,347;stroke-dashoffset:75}25%,75%{stroke-dasharray:17,330}50%{stroke-dasharray:1,347}to{stroke-dasharray:1,347;stroke-dashoffset:423}}\n"]
            },] }
];

class SpinnerInfinityModule {
}
SpinnerInfinityModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [SpinnerInfinityComponent],
                exports: [SpinnerInfinityComponent],
            },] }
];

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerInfinityComponent, SpinnerInfinityModule };
