// spinners-angular/fesm2015/spinners-angular-spinner-circular.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponentWithSecondaryColor } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

class SpinnerCircularComponent extends SpinnerComponentWithSecondaryColor {
    get strokeWidth() {
        return 4 * (this.thickness / 100);
    }
    get dashStyle() {
        return Object.assign({ color: this.color }, (!this.still
            ? { animation: `spinners-angular-circular ${140 / this.speed}s linear infinite` }
            : {}));
    }
}
SpinnerCircularComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-circular',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    [attr.stroke]=\"secondaryColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    stroke-dasharray=\"1, 174\"\r\n    stroke-dashoffset=\"306\"\r\n    stroke-linecap=\"round\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"dashStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-circular{0%{stroke-dashoffset:306}50%{stroke-dasharray:40,134}to{stroke-dasharray:1,174;stroke-dashoffset:132}}@keyframes spinners-angular-circular{0%{stroke-dashoffset:306}50%{stroke-dasharray:40,134}to{stroke-dasharray:1,174;stroke-dashoffset:132}}\n"]
            },] }
];

class SpinnerCircularModule {
}
SpinnerCircularModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [SpinnerCircularComponent],
                exports: [SpinnerCircularComponent],
            },] }
];

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerCircularComponent, SpinnerCircularModule };
