// spinners-angular/fesm2015/spinners-angular-spinner-round-filled.js
import { Component, NgModule } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
import { BrowserModule } from '@angular/platform-browser';

class SpinnerRoundFilledComponent extends SpinnerComponent {
    constructor() {
        super(...arguments);
        this.animations = [
            { r: 4 },
            {
                name: 'spinners-angular-round-filled-inner',
                r: 12.66,
            },
            {
                name: 'spinners-angular-round-filled-center',
                r: 20.32,
            },
            {
                name: 'spinners-angular-round-filled-outer',
                r: 27.5,
            },
        ];
    }
    circleStyle(animation) {
        return {
            opacity: animation.name ? 0.25 : 1,
            transformOrigin: 'center',
            animation: (!animation.name || this.still)
                ? 'none'
                : `${animation.name} ${140 / this.speed}s ease-in-out infinite`,
        };
    }
    radius(animation) {
        return animation.r * (animation.name ? (this.thickness / 100) : 1);
    }
}
SpinnerRoundFilledComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-round-filled',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    *ngFor=\"let animation of animations\"\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"currentColor\"\r\n    [attr.r]=\"radius(animation)\"\r\n    [ngStyle]=\"circleStyle(animation)\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-round-filled-outer{40%,60%{transform:scale(0)}}@keyframes spinners-angular-round-filled-outer{40%,60%{transform:scale(0)}}@-webkit-keyframes spinners-angular-round-filled-center{30%,70%{transform:scale(0)}}@keyframes spinners-angular-round-filled-center{30%,70%{transform:scale(0)}}@-webkit-keyframes spinners-angular-round-filled-inner{20%,80%{transform:scale(0)}}@keyframes spinners-angular-round-filled-inner{20%,80%{transform:scale(0)}}\n"]
            },] }
];

class SpinnerRoundFilledModule {
}
SpinnerRoundFilledModule.decorators = [
    { type: NgModule, args: [{
                imports: [BrowserModule],
                declarations: [SpinnerRoundFilledComponent],
                exports: [SpinnerRoundFilledComponent],
            },] }
];

/**
 * Generated bundle index. Do not edit.
 */

export { SpinnerRoundFilledComponent, SpinnerRoundFilledModule };
