// spinners-angular/esm2015/spinner-dotted/spinner-dotted.component.js
import { Component } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
export class SpinnerDottedComponent extends SpinnerComponent {
    constructor() {
        super(...arguments);
        this.coords = [
            { x: 22, y: -20 },
            { x: 29, y: 0 },
            { x: 22, y: 20 },
            { x: 0, y: 30 },
            { x: -23, y: 20 },
            { x: -30, y: 0 },
            { x: -23, y: -20 },
            { x: 0, y: -30 },
        ];
    }
    get duration() {
        return 200 / this.speed;
    }
    get centerStyle() {
        return !this.still
            ? {
                animation: `spinners-angular-dotted-center ${this.duration}s ease-out infinite`,
                transformOrigin: 'center',
            }
            : { display: 'none' };
    }
    circleStyle(i) {
        return Object.assign({ transform: `translate(${this.coords[i].x}px, ${this.coords[i].y}px)` }, (!this.still
            ? {
                animation: `spinners-angular-dotted-shrink ${this.duration}s cubic-bezier(0, 0.9, 0, 0.9) ${(this.duration / 20) * i}s infinite`
            }
            : {}));
    }
}
SpinnerDottedComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-dotted',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    *ngFor=\"let c of coords; index as i\"\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"currentColor\"\r\n    [attr.r]=\"3 * (thickness / 100)\"\r\n    [ngStyle]=\"circleStyle(i)\"\r\n  />\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"currentColor\"\r\n    [attr.r]=\"6 * (thickness / 100)\"\r\n    [ngStyle]=\"centerStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-dotted-center{0%,15%,85%,to{transform:scale(0)}40%,50%{transform:scale(1)}84%{transform:scale(.45)}}@keyframes spinners-angular-dotted-center{0%,15%,85%,to{transform:scale(0)}40%,50%{transform:scale(1)}84%{transform:scale(.45)}}@-webkit-keyframes spinners-angular-dotted-shrink{50%{transform:translate(0)}}@keyframes spinners-angular-dotted-shrink{50%{transform:translate(0)}}\n"]
            },] }
];
