// spinners-angular/esm2015/spinner-round-outlined/spinner-round-outlined.component.js
import { Component } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
export class SpinnerRoundOutlinedComponent extends SpinnerComponent {
    constructor() {
        super(...arguments);
        this.animations = [
            { r: 2 },
            {
                name: 'spinners-angular-round-outlined',
                r: 14,
            },
            {
                name: 'spinners-angular-round-outlined',
                r: 28,
            },
        ];
    }
    strokeWidth(index) {
        return index ? 3 * (this.thickness / 100) : 4;
    }
    circleStyle(animation) {
        return animation.name && !this.still
            ? { animation: `${animation.name} ${140 / this.speed}s ease-in-out infinite` }
            : {};
    }
}
SpinnerRoundOutlinedComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-round-outlined',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    *ngFor=\"let animation of animations; index as i\"\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    stroke=\"currentColor\"\r\n    [attr.stroke-width]=\"strokeWidth(i)\"\r\n    [attr.r]=\"animation.r\"\r\n    [ngStyle]=\"circleStyle(animation)\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-round-outlined{35%,65%{r:2px;stroke-width:4}}@keyframes spinners-angular-round-outlined{35%,65%{r:2px;stroke-width:4}}\n"]
            },] }
];
