// spinners-angular/esm2015/spinner-round/spinner-round.component.js
import { Component } from '@angular/core';
import { SpinnerComponent } from 'spinners-angular/spinners-utils';
export class SpinnerRoundComponent extends SpinnerComponent {
    get strokeWidth() {
        return 3 * (this.thickness / 100);
    }
    get circleStyle() {
        return {
            animation: !this.still ? `spinners-angular-round ${140 / this.speed}s ease-in-out infinite` : 'none',
            transformOrigin: 'center',
        };
    }
}
SpinnerRoundComponent.decorators = [
    { type: Component, args: [{
                selector: 'sa-spinner-round',
                template: "<svg *ngIf=\"enabled\" fill=\"none\" [ngStyle]=\"svgStyle\" viewBox=\"0 0 66 66\">\r\n  <circle\r\n    cx=\"33\"\r\n    cy=\"33\"\r\n    fill=\"none\"\r\n    r=\"28\"\r\n    stroke=\"currentColor\"\r\n    [attr.stroke-width]=\"strokeWidth\"\r\n    [ngStyle]=\"circleStyle\"\r\n  />\r\n</svg>\r\n",
                styles: ["@-webkit-keyframes spinners-angular-round{27%,73%{r:2px;stroke-width:4}}@keyframes spinners-angular-round{27%,73%{r:2px;stroke-width:4}}\n"]
            },] }
];
