// @proteansoftware/capacitor-start-navigation/dist/plugin.js
var capacitorCapacitorStartNavigation = (function (exports, core) {
    'use strict';

    class StartNavigationPluginWeb extends core.WebPlugin {
        constructor() {
            super({
                name: "StartNavigationPlugin",
                platforms: ["web"],
            });
        }
        async launchMapsApp(options) {
            window.open(`https://www.google.com/maps/dir/?api=1&destination=${options.latitude},${options.longitude}&travelmode=driving`, "_blank");
        }
    }
    const StartNavigation = core.registerPlugin("StartNavigationPlugin", {
        web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.StartNavigationPluginWeb()),
    });

    var web = /*#__PURE__*/Object.freeze({
        __proto__: null,
        StartNavigationPluginWeb: StartNavigationPluginWeb,
        StartNavigation: StartNavigation
    });

    exports.StartNavigation = StartNavigation;
    exports.StartNavigationPluginWeb = StartNavigationPluginWeb;

    Object.defineProperty(exports, '__esModule', { value: true });

    return exports;

})({}, capacitorExports);
